posts/2019-04-11-Rust-1.34.0.md

```diff
@@ -0,0 +1,219 @@
+[arbitrary-tts]: https://github.com/rust-lang/rust/pull/57367
+
+[Procedural macros] in Rust can define custom attributes that they consume.
+Until now, such attributes were restricted to being trees of paths and literals
+according to a specific syntax, like:
+
+```rust
+#[foo(bar)]
+#[foo = "bar"]
+#[foo = 0]
+#[foo(bar = true)]
+#[foo(bar, baz(quux, foo = "bar"))]
+```
+
+Unlike procedural macros, these helper attributes could not accept arbitrary token streams in delimiters,
+so you could not write `#[range(0..10)]` or `#[bound(T: MyTrait)]`.
+Procedural macro crates would instead use strings for specifying syntaxes like this, e.g. `#[range("0..10")]`
+
+With this Rust release, custom attributes `#[attr($tokens)]` [now accept][arbitrary-tts]
+arbitrary token streams in `$tokens`, bringing them on par with macros.
+If you're the author of a procedural macro crate, please check if your custom attributes
+have unnecessary strings in their syntax and if they can be better expressed with token streams.
+
+### `TryFrom` and `TryInto`
+
+[`from_be_bytes`]: https://doc.rust-lang.org/std/primitive.u32.html#method.from_be_bytes
+[never_type]: https://github.com/rust-lang/rust/issues/35121
+[`TryFrom`]: https://doc.rust-lang.org/std/convert/trait.TryFrom.html
+[`TryInto`]: https://doc.rust-lang.org/std/convert/trait.TryInto.html
+[`Infallible`]: https://doc.rust-lang.org/std/convert/enum.Infallible.html
+
+The [`TryFrom`] and [`TryInto`] traits were stabilized to allow fallible type conversions.
+
+For example, the [`from_be_bytes`] and related methods on integer types take arrays,
+but data is often read in via slices. Converting between slices and arrays is tedious to do manually.
+With the new traits, it can be done inline with `.try_into()`.
+
+```rust
+let num = u32::from_be_bytes(slice.try_into()?);
+```
+
+For conversions that cannot fail, such as `u8` to `u32`, the [`Infallible`] type was added.
+This also permits a blanket implementation of `TryFrom` for all existing `From` implementations.
+In the future, we hope to turn `Infallible` into an alias for [the `!` (never) type][never_type].
+
+### `fn before_exec` deprecated in favor of `unsafe fn pre_exec`
+
+[`CommandExt::before_exec`]: https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.before_exec
+[`CommandExt::pre_exec`]: https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.pre_exec
+[ub-possible]: https://github.com/rust-lang/rust/issues/39575#issuecomment-437658766
+[non-dup]: https://github.com/rust-lang/rust/issues/39575#issuecomment-439645949
+
+On Unix-like systems, the function [`CommandExt::before_exec`] allows you to
+schedule a closure to be run before `exec` is invoked.
+
+As the closure provided will be run in the context of the child process after a fork,
+this means that resources, such as file descriptors and memory-mapped regions, may get duplicated.
```

Comment:
This sentence doesn't flow very well (try reading it aloud). Can it be reworded?

Reply:
Changed ever so slightly :)

Reply:
Thank you.